pkg/manifest/render/render.go

```diff
@@ -0,0 +1,274 @@
+		if file.IsDir() {
+			continue
+		}
+		if skipFiles.Has(file.Name()) {
+			continue
+		}
+
+		ipath := filepath.Join(idir, file.Name())
+		iraw, err := ioutil.ReadFile(ipath)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		rraw, err := renderManifest(renderConfig, iraw)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		opath := filepath.Join(odir, file.Name())
+		if err := ioutil.WriteFile(opath, rraw, 0666); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+	}
+
+	agg := utilerrors.NewAggregate(errs)
+	if agg != nil {
+		return fmt.Errorf("error rendering manifests: %v", agg.Error())
+	}
+	return nil
+}
+
+type manifestRenderConfig struct {
+	ReleaseImage string
+}
+
+// renderManifest Executes go text template from `manifestBytes` with `config`.
+func renderManifest(config manifestRenderConfig, manifestBytes []byte) ([]byte, error) {
+	tmpl, err := template.New("manifest").Parse(string(manifestBytes))
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse manifest")
+	}
+
+	buf := new(bytes.Buffer)
+	if err := tmpl.Execute(buf, config); err != nil {
+		return nil, errors.Wrapf(err, "failed to execute template")
+	}
+
+	return buf.Bytes(), nil
+}
+
+// LoadManifests reads manifests out of the given directory and
+// returns a slice of manifests over all loaded manifests.
+func LoadManifests(dir, releaseImage string) ([]manifest.Manifest, error) {
+	tasks, err := loadManifestMetadata(dir, releaseImage)
+	if err != nil {
+		return nil, err
+	}
+
+	var manifests []manifest.Manifest
+	var errs []error
+	for _, task := range tasks {
+		files, err := ioutil.ReadDir(task.idir)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+
+			switch filepath.Ext(file.Name()) {
+			case ".yaml", ".yml", ".json":
+			default:
+				continue
+			}
+
+			p := filepath.Join(task.idir, file.Name())
+			if task.skipFiles.Has(p) {
+				continue
+			}
+
+			raw, err := ioutil.ReadFile(p)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "error reading file %s", file.Name()))
+				continue
+			}
+			if task.preprocess != nil {
+				raw, err = task.preprocess(raw)
+				if err != nil {
+					errs = append(errs, errors.Wrapf(err, "error running preprocess on %s", file.Name()))
+					continue
+				}
+			}
+			ms, err := manifest.ParseManifests(bytes.NewReader(raw))
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "error parsing %s", file.Name()))
+				continue
+			}
+			for i := range ms {
+				ms[i].OriginalFilename = filepath.Base(file.Name())
+			}
+			manifests = append(manifests, ms...)
+		}
+	}
+
+	agg := utilerrors.NewAggregate(errs)
+	if agg != nil {
+		return nil, errors.New(fmt.Sprintf("Error loading manifests from %s: %v", dir, agg.Error()))
+	}
+	return manifests, nil
+}
+
+// LoadManifestsHash returns a hash over all given manifests.
+func LoadManifestsHash(manifests []manifest.Manifest) string {
```

Comment:
nit: This should just be `Hash`.  And maybe `LoadManifests` should just be `Load`?  The package already provides a good deal of qualification, and we don't need to distinguish between other loaders or hashers in the package API.

Reply:
Agree. Not sure how OO "we" are but that naming convention makes sense to me.